File: JavaScript/7js 2/heron.js
/*
|--------------------------------------------------------------------------
| Heron's formula
|--------------------------------------------------------------------------
|
*/

function heron(a, b, c) {
   const x = (a + b + c) / 2
   return Math.sqrt(x * (x - a) * (x - b) * (x - c));
}


const heron = (a, b, c, s = (a + b + c) / 2) => Math.sqrt(s * (s - a) * (s - b) * (s - c));


function heron(...$) {
   $ = [...$]
   let S = ($[0] + $[1] + $[2]) / 2
   return +Math.sqrt(S * (S - $[0]) * (S - $[1]) * (S - $[2])).toFixed(2)
}


function heron(...$) {
   return Math.round((4 * a ** 2 * b ** 2 - (a ** 2 + b ** 2 - c ** 2) ** 2) ** .5 * 25) / 1e2;
}


console.log(heron(3, 4, 5), 6);

File: Python/7py 2/heron.py
"""
|--------------------------------------------------------------------------
| Heron's formula
|--------------------------------------------------------------------------
|
"""

from math import sqrt
def heron(a,b,c):
   x = (a+b+c)/2
   return round(sqrt(x*(x-a)*(x-b)*(x-c)), 2)


def heron(a, b, c):
   s = (a + b + c) / 2
   return round((s * (s - a) * (s - b) * (s - c)) ** 0.5, 2)


def heron(a,b,c):
   return round((1/4)*(4*b*b*c*c-(b*b+c*c-a*a)**2)**(1/2),2)


def heron(a,b,c):
   return round(.25*(4*(b*c)**2-(b*b+c*c-a*a)**2)**.5, 2)


print(heron(3, 4, 5), 6)